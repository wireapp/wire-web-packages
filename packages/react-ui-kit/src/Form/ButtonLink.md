Demo:

```js
import {ButtonLink, Container, Columns, Column, COLOR} from '@wireapp/react-ui-kit';

<Container>
  <Columns>
    <Column>ButtonLink</Column>
    <Column>
      <ButtonLink backgroundColor={COLOR.GREEN}>ButtonLink</ButtonLink>
    </Column>
  </Columns>
  <Columns>
    <Column>Loading ButtonLink</Column>
    <Column>
      <ButtonLink showLoading backgroundColor={COLOR.GREEN}>
        ButtonLink
      </ButtonLink>
    </Column>
  </Columns>
</Container>;
```
