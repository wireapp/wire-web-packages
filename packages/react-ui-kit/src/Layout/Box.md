Demo:

```js
import {Box} from '@wireapp/react-ui-kit';

<Box>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
  vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
  non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.
</Box>;
```
