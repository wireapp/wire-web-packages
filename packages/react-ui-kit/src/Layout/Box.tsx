/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import React from 'react';
import {COLOR} from '../Identity';

export interface BoxProps<T = HTMLDivElement> extends React.HTMLProps<T> {}

const boxStyles: (props: BoxProps) => ObjectInterpolation<undefined> = props => ({
  border: `2px solid ${COLOR.GRAY_LIGHTEN_72}`,
  borderradius: '8px',
  padding: '16px 32px',
});

const Box = (props: BoxProps) => <div css={boxStyles(props)} {...props} />;

export {Box, boxStyles};
