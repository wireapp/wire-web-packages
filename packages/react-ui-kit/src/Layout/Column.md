Demo:

```js
import {Container, Columns, Column, COLOR} from '@wireapp/react-ui-kit';

const ColumnsStyle = {
  marginBottom: '12px',
};

const ColumnStyle = {
  backgroundColor: COLOR.GRAY_LIGHTEN_32,
  border: `1px solid ${COLOR.GRAY_LIGHTEN_24}`,
};

<Container>
  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
  </Columns>
</Container>;
```
