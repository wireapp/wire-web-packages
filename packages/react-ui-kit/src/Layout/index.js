import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export * from './Column';
export * from './Container';
export * from './Content';
export * from './Footer';
export * from './Header';
export * from './StyledApp';
