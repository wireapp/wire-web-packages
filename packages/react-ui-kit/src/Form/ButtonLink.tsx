/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {Loading} from '../Misc';
import {COLOR} from '../Identity';
import {filterProps} from '../util';
import {filterTextProps} from '../Text';
import {ButtonProps, buttonStyle} from './Button';
import {Theme} from '../Layout';

export const buttonLinkStyle: (
  theme: Theme,
  props: ButtonProps<HTMLAnchorElement>,
) => ObjectInterpolation<undefined> = (theme, props) => ({
  ...buttonStyle(theme, props),
  display: 'inline-flex !important',
});

const filterButtonLinkProps = (props: ButtonProps<HTMLAnchorElement>) => {
  return filterProps(filterTextProps(props) as ButtonProps<HTMLAnchorElement>, [
    'backgroundColor',
    'disabled',
    'noCapital',
  ]);
};

export const ButtonLink = ({
  children,
  showLoading,
  loadingColor = COLOR.WHITE,
  ...props
}: ButtonProps<HTMLAnchorElement>) => (
  <a css={theme => buttonLinkStyle(theme, props)} {...filterButtonLinkProps(props)}>
    {showLoading ? <Loading size={30} color={loadingColor} style={{display: 'flex', margin: 'auto'}} /> : children}
  </a>
);
