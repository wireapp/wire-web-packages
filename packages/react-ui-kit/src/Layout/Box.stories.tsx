/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {Meta, StoryObj} from '@storybook/react';

import {Box} from './Box';

import {COLOR} from '../Identity/colors';
import {Text} from '../Text';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Layout/Box',
  decorators: [
    Story => (
      <div style={{padding: '24px', maxWidth: '600px', margin: '0 auto'}}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Box>;

const LOREM_IPSUM = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.`;

export const Default: Story = {
  args: {
    children: LOREM_IPSUM,
  },
};

export const WithCustomPadding: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {padding: '32px'},
  },
};

export const WithCustomBorder: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {border: `2px solid ${COLOR.BLUE}`},
  },
};

export const WithFormattedContent: Story = {
  args: {
    children: (
      <>
        <Text block style={{fontSize: '20px', fontWeight: 'bold', marginBottom: '16px'}}>
          Important Notice
        </Text>
        <Text block style={{marginBottom: '12px'}}>
          {LOREM_IPSUM}
        </Text>
        <Text block style={{color: COLOR.GRAY}}>
          Last updated: 2025-01-01
        </Text>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Text block style={{marginBottom: '16px'}}>
          Outer box content
        </Text>
        <Box style={{backgroundColor: COLOR.GRAY_LIGHTEN_92}}>
          <Text>Inner box content</Text>
        </Box>
      </>
    ),
  },
};

export const Interactive: Story = {
  args: {
    children: (
      <div style={{textAlign: 'center'}}>
        <Text block style={{marginBottom: '16px'}}>
          Click the button below
        </Text>
        <button
          style={{
            backgroundColor: COLOR.BLUE,
            border: 'none',
            borderRadius: '4px',
            color: COLOR.WHITE,
            cursor: 'pointer',
            padding: '8px 16px',
          }}
        >
          Action Button
        </button>
      </div>
    ),
    style: {cursor: 'pointer'},
  },
};
