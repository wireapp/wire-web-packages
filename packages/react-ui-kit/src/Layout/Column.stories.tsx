/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {ReactNode} from 'react';

import {Meta, StoryObj} from '@storybook/react';

import {Column, Columns} from './Column';

import {COLOR} from '../Identity/colors';
import {QueryKeys} from '../mediaQueries';

const meta: Meta<typeof Columns> = {
  component: Columns,
  title: 'Layout/Column',
};

export default meta;
type Story = StoryObj<typeof Columns>;

const Box = ({children}: {children: ReactNode}) => (
  <div
    style={{
      backgroundColor: COLOR.GRAY_LIGHTEN_72,
      border: `1px solid ${COLOR.GRAY_LIGHTEN_48}`,
      borderRadius: '4px',
      padding: '16px',
      textAlign: 'center',
    }}
  >
    {children}
  </div>
);

export const Default: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Column 1</Box>
        </Column>
        <Column>
          <Box>Column 2</Box>
        </Column>
        <Column>
          <Box>Column 3</Box>
        </Column>
      </>
    ),
  },
};

export const TwoColumns: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Column 1</Box>
        </Column>
        <Column>
          <Box>Column 2</Box>
        </Column>
      </>
    ),
  },
};

export const ResponsiveColumns: Story = {
  args: {
    query: QueryKeys.TABLET,
    children: (
      <>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
      </>
    ),
  },
};

export const NestedColumns: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Main Column</Box>
        </Column>
        <Column>
          <Columns>
            <Column>
              <Box>Nested 1</Box>
            </Column>
            <Column>
              <Box>Nested 2</Box>
            </Column>
          </Columns>
        </Column>
      </>
    ),
  },
};
