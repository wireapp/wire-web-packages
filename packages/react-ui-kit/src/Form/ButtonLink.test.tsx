/*
 * Wire
 * Copyright (C) 2019 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import React from 'react';
import {THEME_ID} from '../Layout';
import {matchComponent} from '../test/testUtil';
import {ButtonLink} from './ButtonLink';

describe('"ButtonLink"', () => {
  it('renders', () => matchComponent(<ButtonLink>Submit</ButtonLink>));
  it('renders (dark theme)', () => matchComponent(<ButtonLink>ButtonLink</ButtonLink>, THEME_ID.DARK));
  it('renders in loading state', () => matchComponent(<ButtonLink showLoading>Submit</ButtonLink>));
});
