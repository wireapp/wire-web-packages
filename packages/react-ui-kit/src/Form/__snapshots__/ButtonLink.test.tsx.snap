// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"ButtonLink" renders (dark theme) 1`] = `
.emotion-0 {
  background: #000;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-1 {
  color: var(--text-input-background);
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  font-size: 16px;
  font-weight: 600;
  overflow: hidden;
  text-align: center;
  text-overflow: ellipsis;
  text-transform: none;
  white-space: nowrap;
  border: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-bottom: 16px;
  padding: 0 16px;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  touch-action: manipulation;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
  width: auto;
  border-radius: 16px;
  height: 48px;
  line-height: 48px;
  max-width: 100%;
  min-width: 125px;
  background-color: var(--accent-color);
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: var(--button-primary-hover);
}

.emotion-1:focus {
  border: 1px solid var(--button-primary-focus-border);
}

.emotion-1:active {
  background-color: var(--button-primary-active);
  border: 1px solid var(--button-primary-active-border);
  color: #fff;
}

<div
  className="emotion-0"
>
  <a
    className="emotion-1"
  >
    ButtonLink
  </a>
</div>
`;

exports[`"ButtonLink" renders 1`] = `
.emotion-0 {
  background: #edeff0;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-1 {
  color: var(--text-input-background);
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  font-size: 16px;
  font-weight: 600;
  overflow: hidden;
  text-align: center;
  text-overflow: ellipsis;
  text-transform: none;
  white-space: nowrap;
  border: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-bottom: 16px;
  padding: 0 16px;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  touch-action: manipulation;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
  width: auto;
  border-radius: 16px;
  height: 48px;
  line-height: 48px;
  max-width: 100%;
  min-width: 125px;
  background-color: var(--accent-color);
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: var(--button-primary-hover);
}

.emotion-1:focus {
  border: 1px solid var(--button-primary-focus-border);
}

.emotion-1:active {
  background-color: var(--button-primary-active);
  border: 1px solid var(--button-primary-active-border);
  color: #fff;
}

<div
  className="emotion-0"
>
  <a
    className="emotion-1"
  >
    Submit
  </a>
</div>
`;

exports[`"ButtonLink" renders in loading state 1`] = `
@keyframes animation-0 {
  0% {
    stroke-dashoffset: 251.36;
  }

  100% {
    stroke-dashoffset: 0;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  background: #edeff0;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-1 {
  color: var(--text-input-background);
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  font-size: 16px;
  font-weight: 600;
  overflow: hidden;
  text-align: center;
  text-overflow: ellipsis;
  text-transform: none;
  white-space: nowrap;
  border: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-bottom: 16px;
  padding: 0 16px;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  touch-action: manipulation;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
  width: auto;
  border-radius: 16px;
  height: 48px;
  line-height: 48px;
  max-width: 100%;
  min-width: 125px;
  background-color: var(--accent-color);
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: var(--button-primary-hover);
}

.emotion-1:focus {
  border: 1px solid var(--button-primary-focus-border);
}

.emotion-1:active {
  background-color: var(--button-primary-active);
  border: 1px solid var(--button-primary-active-border);
  color: #fff;
}

.emotion-2 {
  -webkit-animation: animation-0 2400ms ease-in-out infinite,animation-1 2400ms linear -1800ms infinite;
  animation: animation-0 2400ms ease-in-out infinite,animation-1 2400ms linear -1800ms infinite;
  stroke-dasharray: 125.68;
  transform-origin: 50% 50%;
}

<div
  className="emotion-0"
>
  <a
    className="emotion-1"
  >
    <svg
      fill="none"
      height={30}
      strokeWidth="3"
      style={
        {
          "display": "flex",
          "margin": "auto",
        }
      }
      viewBox="0 0 43 43"
      width={30}
    >
      <circle
        cx="21.5"
        cy="21.5"
        r="20"
        stroke="rgba(255, 255, 255, 0.07999999999999996)"
      />
      <circle
        className="emotion-2"
        cx="21.5"
        cy="21.5"
        r="20"
        stroke="#fff"
        strokeDasharray={125.68}
        strokeLinecap="round"
      />
    </svg>
  </a>
</div>
`;
